theme_report/generate_theme_report.py

```diff
@@ -0,0 +1,103 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait # available since 2.4.0
+from selenium.webdriver.support import expected_conditions as EC # available since 2.26.0
+from bs4 import BeautifulSoup
+
+from canvasapi import Canvas, file
+from canvasapi.account import (
+    Account,
+    AccountNotification,
+    AccountReport,
+    Admin,
+    Role,
+    SSOSettings,
+)
+import time
+import pandas as pd
+import io, os, json
+import logging
+
+# Set up ENV
+CONFIG_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), './env.json')
+try:
+    with open(CONFIG_PATH) as env_file:
+        ENV = json.load(env_file)
+except FileNotFoundError:
+    logger.error(f'Configuration file could not be found; please add file "{CONFIG_PATH}".')
+    ENV = dict()
+
+# get the API parameters from ENV
+API_URL = ENV["API_URL"]
+API_KEY = ENV["API_KEY"]
+
+# used for login page
+CANVAS_USER = ENV["CANVAS_USER"]
+CANVAS_PASSWORD = ENV["CANVAS_PASSWORD"]
+
+# Initialize the driver
+driver = webdriver.Chrome()
+
+# Open provided link in a browser window using the driver
+driver.get(API_URL +"/login/saml")
+driver.execute_script(f"document.getElementById('login').value = '{CANVAS_USER}';")
+driver.execute_script(f"document.getElementById('password').value = '{CANVAS_PASSWORD}';")
+loginSubmit = driver.find_element_by_id('loginSubmit')
+loginSubmit.click()
+
+# create an empty dataframe
+global_pd= pd.DataFrame(columns=('account_id', "account_name", "current_theme_name"))
+
+def parse_subaccount_theme(account):
```

Comment:
This was a little confusing.  Up to this point, the main program code is laid out, then it gets interrupted by two function definitions.  They're not referenced until later, when the main program continues after the definitions.

I would put all the main program code into a function, named something like `main`, `run`, `report`, etc.  Then I would call that function at the bottom of the program, preferably in a name-checking conditional, like:

```python
if '__main__' == __name__:
  main()
```

Reply:
restructured code based on the suggestion